### PR TITLE
Support arrays

### PR DESCRIPTION
**Pull request by jbescos**
Relates to this: https://github.com/eclipse-ee4j/jaxrs-api/issues/890

Primitive types are not supported. That is said here in the part '`T` satisfies 1, 3 or 4 above.':
https://github.com/eclipse-ee4j/jaxrs-api/pull/891

### CODE REVIEW

**Comment by senivam** [comment]
shouldn't it be for jakartified spec only?

**Review by jansupol — CHANGES_REQUESTED** [review]
Please make tests for cases 3 and 4, too.
for 4 for instance:
```
    public static class ArrayTestBean {
        private final String value;

        private ArrayTestBean(String value) {
            this.value = value;
        }

        public static ArrayTestBean fromString(String bean) {
            return new ArrayTestBean(bean);
        }

        @Override
        public String toString() {
            return value;
        }
    }

        @Path("/queryBeanArray")
        @GET
        public Response queryBeanArray(@QueryParam(PARAM_NAME) ArrayTestBean[] data) {
            StringBuilder sb = new StringBuilder(data.length);
            for (ArrayTestBean d : data) {
                sb.append(d);
            }
            return Response.ok(sb.toString()).build();
        }

        @Path("/queryBeanList")
        @GET
        public Response queryBeanList(@QueryParam(PARAM_NAME) List<ArrayTestBean> data) {
            StringBuilder sb = new StringBuilder(data.size());
            for (ArrayTestBean d : data) {
                sb.append(d);
            }
            return Response.ok(sb.toString()).build();
        }

    @Test
    public void queryBean() {
        Response expected = target("/test/queryBeanList").queryParam(PARAM_NAME, "1", "2").request().get();
        Response arrayResponse = target("/test/queryBeanArray").queryParam(PARAM_NAME, "1", "2").request().get();
        verifyStr(expected, arrayResponse);
    }
```

**Comment by jansupol** [comment]
> shouldn't it be for jakartified spec only?

For 3.1, it would be a MUST. For 2.x, it is an enhancement.

File: connectors/jdk-connector/src/test/java/org/glassfish/jersey/jdk/connector/internal/ArrayParamConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/*
```
[the remark this answers is not in the document]
Comment (jbescos) [reply connectors/jdk-connector/src/test/java/org/glassfish/jersey/jdk/connector/internal/ArrayParamConverterTest.java]: Good to know. By default I was adding tests in JDK connector.

**Comment by jbescos** [comment]
Primitive types are supported in the last commit

**Review by jansupol — APPROVED** [review]

**Review by senivam — APPROVED** [review]

**Pull request merged**